README.md

```diff
@@ -1,144 +1,104 @@
 
-_This library is in flux while we work on implementing sourcepoint and supporting CCPA and TCFv2. If you only need TCFv1, the instructions below still apply. If you also need CCPA, or you don't know what any of this means, ping us in chat!_
+Welcome to the Consent Management Platform, a library to manage privacy frameworks across \*.theguardian.com, from messaging to reporting the privacy state. All exports include Typescript definitions.
 
-[![Generic badge](https://img.shields.io/badge/google-chat-259082.svg)](https://chat.google.com/room/AAAAhlhgDTU)
+## What utilities does this offer?
 
----
+### TCF and CCPA privacy frameworks
 
-Welcome to the Consent Management Platform, a library of useful utilities for managing consent state across \*.theguardian.com. All exports include Typescript definitions.
-
-## What useful utilities does this offer?
+If you need to use the TCF or CCPA privacy frameworks, you can inititalise them on your page using the `init` function.
 
 ### Consent notifications
 
-If you need to conditionally run some code based on a user's consent state you can use the two functions `onGuConsentNotification` and `onIabConsentNotification`.
+If you need to conditionally run some code based on a user's consent state you can use the function `onConsentNotification`.
 
-#### onGuConsentNotification
+## What is the API?
 
-This function takes 2 arguments, the first is the purpose name (a `string`) that is relevant to the code you're running eg. "functional" OR "performance", and the second is a callback (a `function`).
+[init](#init)<br />
+[onConsentNotification](#onconsentnotification)<br />
+[checkUiWillShow](#checkuiwillshow)<br />
+[showPrivacyManager](#showprivacymanager)
 
-When `onGuConsentNotification` is called it will execute the callback immediately, passing it a single argument (a `boolean` or `null`) which indicates the user's consent state at that time for the given purpose name.
+### init
 
-The `cmp` module also listens for subsequent changes to the user's consent state (eg. if a user saves an update to their consent via the CMP modal), if this happens it will re-execute the callback, passing it a single argument (a `boolean` or `null`) which inidicates the user's updated consent state for the given purpose name.
+`init(options: InitOptions): void`
 
-**Example:**
+Calling `init` will add the TCF or the CCPA privacy framework to the page, depending on the configuration options received. We refer to these as TCF mode and CCPA mode, respectively. This function needs to be run before any other API call.
 
-```js
-import { onGuConsentNotification } from '@guardian/consent-management-platform';
+The configuration object that it requires is:
 
-onGuConsentNotification('functional', (functionalConsentState) => {
-	console.log(functionalConsentState); // true || false || null
-});
 ```
+interface InitOptions {
+	isInUS: boolean;
+}
+```
+
+If `InitOptions.isInUS` is missing, `init` will default to running in TCF mode.
```

Comment:
this feels dangerous to me – would it be safer to force people to be explicit?

Reply:
I don't think it's that dangerous because in TCFv2 mode nothing fires unless given explicit consent but I'm happy to change it.
What do you suggest happens if the `isInUS` flag is missing?

Reply:
I was thinking that because we have to show the CCPA in California, we shouldn't give people the chance to not consider where their code will run?

how about we throw an error, as well as TS failing?

Reply:
Throwing an error seems good since projects that don't use TS, such as frontend, won't see a TS error.